File: automovil_g31/clases/persona.py
class Persona:
    def __init__(self, nombre, id, l_automoviles=None):
        self.nombre = nombre
        self.id = id
        self.l_automoviles = l_automoviles if l_automoviles is not None else []

    @property
    def nombre(self):
        return self._nombre.upper()

    @nombre.setter
    def nombre(self, value):
        if not value or len(value.strip()) < 6:
            raise ValueError("Revise que el nombre tenga más de 6 caracteres y no sea nulo o vacío.")
        self._nombre = value

    @property
    def id(self):
        return self._id

    @id.setter
    def id(self, value):
        if not value or len(value.strip()) < 6:
            raise ValueError("Revise que el ID tenga más de 6 caracteres y no sea nulo o vacío. No se pudo crear el objeto.")
        self._id = value

    @property
    def l_automoviles(self):
        return self._l_automoviles

    @l_automoviles.setter
    def l_automoviles(self, value):
        if not isinstance(value, list):
            raise ValueError("l_automoviles debe ser una lista.")
        self._l_automoviles = value

    def agregar_automovil(self, automovil):
        if not isinstance(automovil, Automovil):
            raise ValueError("El objeto agregado debe ser una instancia de Automovil.")
        self._l_automoviles.append(automovil)

    def eliminar_automovil(self, automovil):
        try:
            self._l_automoviles.remove(automovil)
        except ValueError:
            raise ValueError("El automóvil no se encuentra en la lista.")

    def eliminar_automovil_por_indice(self, index):
        if index < 0 or index >= len(self._l_automoviles):
            raise IndexError("Índice fuera de rango.")
        self._l_automoviles.pop(index)

    def mostrar_automoviles(self):
        if not self._l_automoviles:
            print("La persona no tiene automóviles.")
            return

        print(f"Automóviles de {self.nombre}:")
        for auto in self._l_automoviles:
            print(f"- Marca: {auto.marca}, Modelo: {auto.modelo}, Año: {auto.ano}")

File: automovil_g31/clases/persona.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace automovil_G3.clases
{
    class Persona
    {
        private string nombre;
        private string id;
        private List<Automovil> l_automoviles; // Implementación de la relación

        // Constructor 1: Va a pedir nombre e id y crea lista en blanco
        public Persona(string nombre, string id)
        {
            Nombre = nombre;
            Id = id;
            // Creación lista vacía
            l_automoviles = new List<Automovil>();
        }

        // Constructor 2: Va a pedir nombre, id y lista de automóviles
        public Persona(string nombre, string id, List<Automovil> l_automoviles)
        {
            Nombre = nombre;
            Id = id;
            L_automoviles = l_automoviles;
        }

        public string Nombre
        {
            get => nombre.ToUpper();
            set
            {
                if (string.IsNullOrEmpty(value) || string.IsNullOrWhiteSpace(value) || value.Length < 6)
                    throw new Exception("Revise que el nombre tenga más de 6 caracteres y no sea nulo o vacío.");
                else
                    nombre = value;
            }
        }

        public string Id
        {
            get => id;
            set
            {
                if (string.IsNullOrEmpty(value) || string.IsNullOrWhiteSpace(value) || value.Length < 6)
                    throw new Exception("Revise que el ID tenga más de 6 caracteres y no sea nulo o vacío. No se pudo crear el objeto.");
                else
                    id = value;
            }
        }

        internal List<Automovil> L_automoviles
        {
            get => l_automoviles;
            set => l_automoviles = value;
        }

        // Método para añadir un automóvil a la lista
        public void AgregarAutomovil(Automovil automovil)
        {
            if (automovil == null)
                throw new Exception("El automóvil no puede ser nulo.");
            l_automoviles.Add(automovil);
        }

        // Método para eliminar un automóvil de la lista por objeto
        public void EliminarAutomovil(Automovil automovil)
        {
            if (!l_automoviles.Remove(automovil))
                throw new Exception("El automóvil no se encuentra en la lista.");
        }

        // Método para eliminar un automóvil de la lista por índice
        public void EliminarAutomovil(int index)
        {
            if (index < 0 || index >= l_automoviles.Count)
                throw new Exception("Índice fuera de rango.");
            l_automoviles.RemoveAt(index);
        }

        // Método para mostrar información de todos los automóviles
        public void MostrarAutomoviles()
        {
            if (l_automoviles.Count == 0)
            {
                Console.WriteLine("La persona no tiene automóviles.");
                return;
            }

            Console.WriteLine("Automóviles de " + Nombre + ":");
            foreach (var auto in l_automoviles)
            {
                Console.WriteLine($"- Marca: {auto.Marca}, Modelo: {auto.Modelo}, Año: {auto.Ano}");
            }
        }
    }
}
